File: Web/chanllenge/src/main/java/com/diegomazega/chanllenge/domain/Pessoa.java
package com.diegomazega.chanllenge.domain;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Objects;
import java.util.Set;

import javax.persistence.CascadeType;
import javax.persistence.CollectionTable;
import javax.persistence.Column;
import javax.persistence.ElementCollection;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.MapsId;
import javax.persistence.OneToMany;
import javax.persistence.OneToOne;

@Entity
public class Pessoa implements Serializable {
	private static final long serialVersionUID = 1L;
	
	@Id
	@GeneratedValue(strategy=GenerationType.IDENTITY)
	private Long id;
	
	@Column(unique = true)
	private String nome;
	
	@Column()
	private Integer idade;
	
	@ElementCollection
	@CollectionTable(name="telefone")
	private Set<String> telefones = new HashSet<>();
	
	@OneToMany(mappedBy = "pessoa", cascade = CascadeType.ALL)
	private List<Endereco> enderecos = new ArrayList<>();
	
	@OneToOne(cascade = CascadeType.ALL)
	@JoinColumn(name = "funcionario_id")
	@MapsId
	private Funcionario funcionario;

	public Pessoa() {}
	
	public Pessoa(Long id, String nome, Integer idade, Set<String> telefones) {
		super();
		this.id = id;
		this.nome = nome;
		this.idade = idade;
		this.telefones = telefones;
	}

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getNome() {
		return nome;
	}

	public void setNome(String nome) {
		this.nome = nome;
	}

	public Integer getIdade() {
		return idade;
	}

	public void setIdade(Integer idade) {
		this.idade = idade;
	}

	public List<Endereco> getEnderecos() {
		return enderecos;
	}

	public void setEnderecos(List<Endereco> endereco) {
		this.enderecos = endereco;
	}

	public Funcionario getFuncionario() {
		return funcionario;
	}

	public void setFuncionario(Funcionario funcionario) {
		this.funcionario = funcionario;
	}

	public Set<String> getTelefones() {
		return telefones;
	}

	public void setTelefones(Set<String> telefones) {
		this.telefones = telefones;
	}
	
	@Override
	public int hashCode() {
		return Objects.hash(id);
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Pessoa other = (Pessoa) obj;
		return Objects.equals(id, other.id);
	}
	
	
	
}

File: Frontend/viasoft/lib/Models/Pessoa.dart
// ignore_for_file: file_names, unnecessary_getters_setters
// ignore_for_file: avoid_print

import 'package:viasoft/Models/Endereco.dart';
import 'package:viasoft/Models/Funcionario.dart';

class Pessoa{
  int _id;
  String _nome;
  int _idade;
  List<String> _telefones;
  List<Endereco> _enderecos;
  Funcionario _funcionario;
 
 Pessoa(this._id, this._nome, this._idade, this._telefones, this._enderecos, this._funcionario);

 int get id => _id;

 set id(int value) => _id = value;

  get nome => _nome;

 set nome( value) => _nome = value;

  get getIdade => _idade;

 set setIdade( idade) => _idade = idade;

  get telefones => _telefones;

 set telefones( value) => _telefones = value;

  get enderecos => _enderecos;

 set enderecos( value) => _enderecos = value;

  get funcionario => _funcionario;

 set funcionario( value) => _funcionario = value;


}